uPortal-content/uPortal-content-portlet/src/main/java/org/apereo/portal/portlet/dao/jpa/PortletDefinitionImpl.java

```diff
@@ -714,18 +665,59 @@ public EntityIdentifier getEntityIdentifier() {
 
     @Override
     public PortletLifecycleState getLifecycleState() {
+
+        /*
+         * NB:  A portlet definition may have a current lifecycle
+         * state of null if it has not yet been completely created.
+         */
+        PortletLifecycleState rslt = null;
+
         final Date now = new Date();
-        if (parameters.containsKey(PortletLifecycleState.MAINTENANCE_MODE_PARAMETER_NAME)) {
-            return PortletLifecycleState.MAINTENANCE;
-        } else if (expirationDate != null && expirationDate.before(now)) {
-            return PortletLifecycleState.EXPIRED;
-        } else if (publishDate != null && publishDate.before(now)) {
-            return PortletLifecycleState.PUBLISHED;
-        } else if (approvalDate != null && approvalDate.before(now)) {
-            return PortletLifecycleState.APPROVED;
-        } else {
-            return PortletLifecycleState.CREATED;
+        for (IPortletLifecycleEntry entry : lifecycleEntries) {
```

Comment:
That said, I don't mind looking at using a Stream in this particular case.  I simply want to be clear that there are a few things we require -- e.g. WCAG 2.0 Level AA, lack of security vulnerabilities, _etc_. -- and then there are a great many things we value.